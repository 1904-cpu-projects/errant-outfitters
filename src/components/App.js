import React from "react";

class App extends React.Component {
  // gotta see that HR!
  render() {
    return (
      <div>
        <hr />
      </div>
    );
  }
}

export default App;
