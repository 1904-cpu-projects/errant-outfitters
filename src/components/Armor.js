import React from 'react';
import { connect } from 'react-redux';
import { SingleProduct } from './SingleProduct';
import MenuBar from './MenuBar';

export function Products({ products }) {
  const armor = products.productList.filter(a => a.catagory === 'armor');
  //function to make tiles from SingleProduct component

  return (
    <div>
      <MenuBar />
      <div className="products-list">
        {armor.map(p => {
          return <SingleProduct key={p.id} product={p} />;
        })}
      </div>
    </div>
  );
}

const mapStateToProps = state => ({
  products: state.products,
});

export default connect(mapStateToProps)(Products);
