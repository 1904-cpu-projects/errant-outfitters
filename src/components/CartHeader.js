import React from 'react';
import { Link } from 'react-router-dom';

// numberItems needs to just be a number
// pull in from redux store CartStore I think
export function CartHeader({ cart }) {
  const cartItems = cart ? cart.length : 0;
  return (
    <div>
      <Link to="/user/profile">Cart</Link> | {cartItems}
    </div>
  );
}
