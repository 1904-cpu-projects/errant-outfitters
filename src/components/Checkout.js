import React from 'react';
import PropTypes from 'prop-types';
import { Elements, StripeProvider } from 'react-stripe-elements';
import { connect } from 'react-redux';

import StripCard from './StripeCard';
import { getCart } from '../storeReducers/cartReducer';
import { listProductsThunk } from '../actions/productActions';

class Checkout extends React.Component {
  constructor() {
    super();
    this.state = {
      total: 0,
      transactionComplete: false,
      transacitons: [],
    };
    this.updateCart = this.updateCart.bind(this);
  }

  calcTotal(cart) {
    const total = cart.items.reduce((acc, item) => {
      return acc + item.product.cost * item.quantity;
    }, 0);
    this.setState({ total: total });
  }

  updateCart(data) {
    console.log(data);
    this.props.getCart();
    this.props.listProductsThunk();
    this.setState({ transactionComplete: true, transactions: [...data] });
  }

  componentDidMount() {
    const { cart } = this.props;
    this.calcTotal(cart);
  }

  render() {
    if (!this.state.transactionComplete) {
      return (
        <div className="checkout">
          <p>Would you like to complete the purchase?</p>
          <div>
            <StripeProvider apiKey={process.env.STRIPE_API}>
              <Elements>
                <StripCard updateCart={this.updateCart} />
              </Elements>
            </StripeProvider>
          </div>
          <div>
            <h2>Your total for this purchase is ${this.state.total}</h2>
            <p>
              You don&apos;t even need to worry about shipping information
              <br />
              We have mages on standby to teleport your items directly to you!
            </p>
          </div>
        </div>
      );
    } else {
      return (
        <div>
          <h2>Your transaction is complete!</h2>
          <div>
            Total cost: {this.state.transactions[0].totalCost}
            <br />
            Transaction #: {this.state.transactions[0].id}
            <br />
            Your Items:
            <br />
            {this.state.transactions.map((item, idx) => (
              <p key={idx}>
                {item.product.name} | Quantity: {item.quantity}
              </p>
            ))}
          </div>
        </div>
      );
    }
  }
}

Checkout.propTypes = {
  cart: PropTypes.object,
  getCart: PropTypes.func,
  listProductsThunk: PropTypes.func,
};

const mapStateToProps = state => ({
  cart: state.cart,
});

const mapDispatchToProps = dispatch => ({
  getCart: () => dispatch(getCart()),
  listProductsThunk: () => dispatch(listProductsThunk()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Checkout);
