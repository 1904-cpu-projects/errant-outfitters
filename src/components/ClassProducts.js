import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { SingleProduct } from './SingleProduct';

export function Products({ products, user }) {
  const classProducts = products.productList.filter(
    a => a.class === user.class,
  );
  //function to make tiles from SingleProduct component
  if (classProducts.length !== 0) {
    return (
      <div>
        <div className="products-list">
          {classProducts.map(p => {
            return <SingleProduct key={p.id} product={p} />;
          })}
        </div>
      </div>
    );
  } else {
    return (
      <div>
        <h2 className="products-list">
          Please register/login to see your class recommended products
        </h2>
      </div>
    );
  }
}

Products.propTypes = {
  productList: PropTypes.array,
  user: PropTypes.object,
};

const mapStateToProps = state => ({
  products: state.products,
  user: state.user,
});

export default connect(mapStateToProps)(Products);
