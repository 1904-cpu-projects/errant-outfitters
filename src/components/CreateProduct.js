import React from 'react';
import { connect } from 'react-redux';
import { postProductThunk } from '../actions/productActions';
import PropTypes from 'prop-types';

class CreateProduct extends React.Component {
  constructor() {
    super();
    this.state = {
      name: '',
      description: '',
      image: '',
      category: '',
      stock: '',
      cost: '',
    };

    this.onHandle = this.onHandle.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.onCategorySelection = this.onCategorySelection.bind(this);
    this.onInstockSelection = this.onInstockSelection.bind(this);
  }

  onHandle(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  onCategorySelection(event) {
    this.setState({
      category: event.target.value,
    });
  }

  onInstockSelection(event) {
    this.setState({
      inStock: event.target.value,
    });
  }

  onSubmit(event) {
    event.preventDefault();
    this.props.postProductThunk(this.state);
    window.location.hash = '/#/';
  }

  render() {
    return (
      <div className="create-product-form">
        <h1>Add a new product to the set</h1>
        <form onSubmit={this.onSubmit}>
          <label htmlFor="name">Name: </label>
          <input
            onChange={this.onHandle}
            type="text"
            name="name"
            value={this.state.name}
          />

          <label htmlFor="description">Description: </label>
          <textarea
            onChange={this.onHandle}
            type="text"
            value={this.state.description}
            name="description"
          />

          <label htmlFor="image">Image: </label>
          <input
            onChange={this.onHandle}
            type="text"
            value={this.state.image}
            name="image"
          />

          <label htmlFor="inStock">inStock: </label>
          <select onChange={this.onInstockSelection} value={this.state.inStock}>
            <option value="false">false</option>
            <option value="true">true</option>
          </select>

          <label htmlFor="stock">Stock: </label>
          <input
            onChange={this.onHandle}
            name="stock"
            type="number"
            min="0"
            value={this.state.stock}
          />

          <label htmlFor="cost">Cost: </label>
          <input
            onChange={this.onHandle}
            name="cost"
            type="number"
            min="0"
            value={this.state.cost}
          />

          <label htmlFor="category">Category: </label>
          <select
            onChange={this.onCategorySelection}
            value={this.state.category}
          >
            <option key="0" value="armor">
              Armour
            </option>
            <option key="1" value="weapon">
              Weapon
            </option>
            <option key="2" value="potion">
              Potion
            </option>
          </select>
          <button type="submit">Submit</button>
        </form>
      </div>
    );
  }
}

CreateProduct.propTypes = {
  postProductThunk: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => {
  return {
    postProductThunk: thing => dispatch(postProductThunk(thing)),
  };
};

export default connect(
  null,
  mapDispatchToProps,
)(CreateProduct);
