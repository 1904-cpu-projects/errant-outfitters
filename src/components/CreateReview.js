import React from "react";
import { connect } from "react-redux";
import { postReview } from "../actions/reviewActions";

class CreateReview extends React.Component {
  constructor() {
    super();
    this.state = {
      title: "",
      email: "", //make the default author the emal of the user
      body: ""
    };
    this.onHandle = this.onHandle.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onHandle(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  onSubmit(event) {
    event.preventDefault();
    this.props.postReview(this.state);
  }

  render() {
    console.log("REVIEW FORM", this.props);
    return (
      <div>
        <form onSubmit={this.onSubmit}>
          <label htmlFor="title">Title: </label>
          <input
            name="title"
            value={this.state.title}
            type="text"
            onChange={this.onHandle}
          />

          <label htmlFor="email">Email: </label>
          <input
            name="email"
            type="email"
            value={this.state.email}
            onChange={this.onHandle}
          />
          <label htmlFor="body">Please enter your review: </label>
          <input
            name="body"
            type="text"
            value={this.state.body}
            onChange={this.onHandle}
          />
          <button type="submit">Submit</button>
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    postReview: stuff => dispatch(postReview(stuff))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(CreateReview);
