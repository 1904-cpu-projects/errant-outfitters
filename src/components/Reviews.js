import React from "react";
import { listReviews } from "../actions/reviewActions";
import { connect } from "react-redux";

//for this reviews page, we need to properly associate the Product with the reviews
class Reviews extends React.Component {
  componentDidMount() {
    this.props.listReviews();
  }
  render() {
    const { reviews } = this.props.reviews;
    const filteredReviews = reviews.filter(
      item => item.productId === this.props.productId
    );
    //reviews need to be filtered based on the product id
    console.log("REVIEWS", this.props);
    console.log("FILTERED reviews", filteredReviews);
    return (
      <div>
        <footer>
          <h4>Reviews for the {this.props.product.name}</h4>
          <div>
            {filteredReviews.map(i => (
              <div className="reviewDiv" key={i.id}>
                <h4>{i.title}</h4>
                <h5>{i.body}</h5>
              </div>
            ))}
          </div>
        </footer>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  reviews: state.reviews
});

const mapDispatchToProps = dispatch => {
  return {
    listReviews: () => dispatch(listReviews())
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Reviews);
