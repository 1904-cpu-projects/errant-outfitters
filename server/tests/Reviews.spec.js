import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { Reviews } from '../../src/components/Reviews';

configure({ adapter: new Adapter() });

const mockUser = {
  id: '456',
};
const mockReviews = {
  reviews: [
    { title: 'review1', body: 'stuff', productId: '123', userId: '456' },
    { title: 'review1', body: 'more stuff', productId: 'abc', userId: 'def' },
  ],
};
const mockFunction = jest.fn();
const mockDeleteReview = jest.fn();

describe('<Reviews/>', () => {
  it('renders the Reviews component present', () => {
    const wrapper = shallow(
      <Reviews
        listReviews={mockFunction}
        user={mockUser}
        reviews={mockReviews}
        deleteReview={mockDeleteReview}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Calls the correct function to delete a review', () => {
    const wrapper = shallow(
      <Reviews
        listReviews={mockFunction}
        user={mockUser}
        reviews={mockReviews}
        deleteReview={mockDeleteReview}
      />,
    );
    const deleteButton = wrapper.find('.review-delete');
    deleteButton.simulate('click');
    expect(mockDeleteReview).toHaveBeenCalledTimes(1);
    expect(mockDeleteReview).toHaveBeenCalledWith({
      title: 'review1',
      body: 'stuff',
      productId: '123',
      userId: '456',
    });
  });
});
