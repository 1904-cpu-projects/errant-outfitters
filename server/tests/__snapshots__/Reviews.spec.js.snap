// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reviews/> renders the Reviews component present 1`] = `
<div>
  <footer>
    <div>
      <div
        className="reviewDiv"
      >
        <h4>
          review1
        </h4>
        <h5>
          stuff
        </h5>
        <button
          className="review-delete"
          onClick={[Function]}
          type="submit"
        >
          Delete Review
        </button>
      </div>
    </div>
  </footer>
</div>
`;
